tgui/packages/tgui/interfaces/Manufacturer/index.tsx

```diff
@@ -0,0 +1,321 @@
+          label="Owner"
+        >
+          {data.card_owner}
+        </LabeledList.Item>
+        <LabeledList.Item
+          label="Balance"
+        >
+          {formatMoney(data.card_balance)}⪽
+        </LabeledList.Item>
+      </LabeledList>
+    </Section>
+  );
+};
+
+export const Manufacturer = (_, context) => {
+  const { act, data } = useBackend<ManufacturerData>(context);
+  const [repeat, toggleRepeatVar] = useSharedState(context, "repeat", data.repeat);
+  const [search, setSearchData] = useLocalState(context, "query", "");
+  const [swap, setSwappingMaterial] = useLocalState(context, "swap", null);
+  const wirePanel:MaintenencePanel = {
+    indicators: data.indicators,
+    wires: data.wires,
+    wire_bitflags: data.wire_bitflags,
+  };
+  let toggleRepeat = () => {
+    act("repeat");
+    toggleRepeatVar(!repeat);
+  };
+  let swapPriority = (materialID: string) => {
+    if (swap === null) {
+      setSwappingMaterial(materialID);
+    }
+    else if (swap === materialID) {
+      setSwappingMaterial(null);
+    }
+    else {
+      act("material_swap", { "resource_1": swap, "resource_2": materialID });
+      setSwappingMaterial(null);
+    }
+  };
+  const all_blueprint_lists = [
+    data.available_blueprints,
+    data.downloaded_blueprints,
+    data.recipe_blueprints,
+    data.hidden_blueprints,
+  ];
+  const all_blueprint_list_strings = [
+    "available",
+    "download",
+    "drive_recipes",
+    "hidden",
+  ];
+  /*
+    Converts the blueprints we get into one larger list sorted by category.
+    This is done here instead of sending one big list to reduce the amount of times we need to refresh static data.
+  */
+  let blueprints_by_category:Record<string, Manufacturable[]> = {};
+  let test = "";
+  for (let category_index = 0; category_index < data.all_categories.length; category_index++) {
+    let category = data.all_categories[category_index];
+    blueprints_by_category[category] = [];
+    for (let blueprint_index = 0; blueprint_index < all_blueprint_lists.length; blueprint_index++) {
+      if (!data.hacked && (all_blueprint_list_strings[blueprint_index] === "hidden")) {
+        continue;
+      }
+      let blueprint_list = all_blueprint_lists[blueprint_index];
+      if (blueprint_list[category] === undefined) {
+        continue;
+      }
+      for (let blueprint of blueprint_list[category]) {
+        if (blueprint.name.toLowerCase().includes(search)) {
+          blueprints_by_category[blueprint.category].push(blueprint);
+        }
+      }
+    }
+  }
+
+
+  // Get a Manufacturable from a QueueBlueprint using its type, category, and name.
+  let getBlueprintFromQueueData = (queueData:QueueBlueprint) => {
+    return blueprints_by_category[queueData.category].find((key) => (key.name === queueData.name));
+  };
+
+  return (
+    <Window width={1200} height={600} title={data.fabricator_name}>
+      <Window.Content scrollable>
+        <Stack>
+          <Stack.Item width="80%">
```

Comment:
```suggestion
          <Stack.Item grow>
```
Then set the sidebar to a fixed width. The sidebar is going to be very sensitive to width changes, better to make it static and then have the list of items fill the rest of the space.